server/handler/fetcher.go

```diff
@@ -37,26 +39,50 @@ type ConfigFetcher struct {
 }
 
 func (cf *ConfigFetcher) ConfigForRepositoryBranch(ctx context.Context, client *github.Client, owner, repository, branch string) FetchedConfig {
+	retries := 0
+	delay := 1 * time.Second
+	ticker := time.NewTicker(delay)
+	defer ticker.Stop()
+	for {
+		c, err := cf.Loader.LoadConfig(ctx, client, owner, repository, branch)
+		fc := FetchedConfig{
+			Source: c.Source,
+			Path:   c.Path,
+		}
 
-	c, err := cf.Loader.LoadConfig(ctx, client, owner, repository, branch)
-	fc := FetchedConfig{
-		Source: c.Source,
-		Path:   c.Path,
-	}
+		if err != nil {
+			if !os.IsTimeout(err) {
+				fc.LoadError = err
+				return fc
+			}
 
-	switch {
-	case err != nil:
-		fc.LoadError = err
-		return fc
-	case c.IsUndefined():
-		return fc
-	}
+			retries++
+			if retries > 3 {
+				fc.LoadError = err
+				return fc
+			}
 
-	var pc policy.Config
-	if err := yaml.UnmarshalStrict(c.Content, &pc); err != nil {
-		fc.ParseError = err
-	} else {
-		fc.Config = &pc
+			ticker.Reset(delay)
```

Comment:
Let's use `time.After` here instead of manipulating a ticker - I think it will be easier to understand and I don't think there are any performance concerns with allocating a new timer if we're about to sleep for multiple seconds. Go 1.23 also fixed the issue with timers not being GC'd until after they fire.

Reply:
Sounds good, I guess I got used to using ticker before 1.23 :relaxed: